manifests/config.pp

```diff
@@ -44,9 +44,21 @@
   }
 
   if $::foreman::use_foreman_service {
-    systemd::dropin_file { 'installer.conf':
-      unit    => "${::foreman::foreman_service}.service",
-      content => template('foreman/foreman.service-overrides.erb'),
+    $listen_socket = $foreman::foreman_service_bind ? {
+      Stdlib::IP::Address::V6 => "[${foreman::foreman_service_bind}]:${foreman::foreman_service_port}",
+      default                 => "${foreman::foreman_service_bind}:${foreman::foreman_service_port}",
+    }
+
+    systemd::dropin_file { 'foreman-socket':
+      filename => 'installer.conf',
+      unit     => "${::foreman::foreman_service}.socket",
+      content  => "[Service]\nListenSocket=\nListenSocket=${listen_socket}\n",
```

Comment:
This is harder to read than say a template or multiline content, but I think what I am seeing is this example output:

```
[Service]
ListenSocket=
ListenSocket=127.0.0.1:3000
```

If so, what is the empty ListenSocket doing?

Reply:
It clears the built in from `foreman.socket`, similar to how we use `ExecStart=`. Otherwise it's in addition to since `ListenSocket` can be repeated.

I first had it without the `ListenSocket=` and then it was short enough but now I agree it may be a bit less readable. Because it's short, I'll have a look at using heredoc instead.

Reply:
And for completeness, `systemctl cat foreman.socket`

```text
# /etc/systemd/system/foreman.socket
[Unit]
Description=Puma HTTP Server Accept Sockets

[Socket]
ListenStream=3000

# AF_UNIX domain socket
# SocketUser, SocketGroup, etc. may be needed for Unix domain sockets
# ListenStream=/run/puma.sock

# Socket options matching Puma defaults
NoDelay=true
ReusePort=true
Backlog=1024

[Install]
WantedBy=sockets.target

# /etc/systemd/system/foreman.socket.d/installer.conf
[Service]
ListenSocket=
ListenSocket=127.0.0.1:3000
```